handlers/deploy.go

```diff
@@ -207,11 +222,27 @@ func makeDeploymentSpec(request types.FunctionDeployment, existingSecrets map[st
 							SecurityContext: &corev1.SecurityContext{
 								ReadOnlyRootFilesystem: &request.ReadOnlyRootFilesystem,
 							},
+							VolumeMounts: []apiv1.VolumeMount{
```

Comment:
This adds a volume even if one isn't configured, so you should set up the slice outside of this piece of code. ideally in a method to create it.